core/src/test/java/org/infinispan/conflict/impl/OperationsDuringMergeConflictTest.java

```diff
@@ -182,13 +180,20 @@ public void handleFromRemoteSite(String origin, XSiteReplicateCommand command, R
 
       @Override
       public void handle(CacheRpcCommand command, Reply reply, DeliverOrder order) {
-         if (command instanceof StateResponseCommand) {
-            try {
-               latch.await();
-            } catch (InterruptedException ignore) {
-            }
-         }
+         if (command instanceof StateResponseCommand)
+            awaitLatch(latch);
          delegate.handle(command, reply, order);
       }
    }
+
+   private void awaitLatch(CountDownLatch latch) {
+      try {
+         // Timeout has to be large enough to allow for rebalance and subsequent operations, so we double the
+         // rebalance timeout. Timeout necessary as for some reason the latch is not always counted down in the Handler
```

Comment:
@ryanemerson I'm now seeing another failure on my machine:

    testGetConflictsDuringStateTransfer[DIST_SYNC](org.infinispan.conflict.impl.ConflictManagerTest)  Time elapsed: 0.968 sec  <<< FAILURE!
    org.testng.TestException: 
    Method org.infinispan.test.fwk.NamedTestMethod@8bf055db should have thrown an exception of class java.lang.IllegalStateException